tests/test_full_workflow.py

```diff
@@ -443,3 +445,84 @@ async def test_failing_workflow_JobExecutionError(
         assert "JOB ERROR" in job_status_data["log"]
         assert "CANCELLED" in job_status_data["log"]
         assert "\\n" not in job_status_data["log"]
+
+
+async def test_non_python_task(
+    client,
+    MockCurrentUser,
+    testdata_path,
+    project_factory,
+    dataset_factory,
+    resource_factory,
+    tmp777_path,
+):
+    async with MockCurrentUser(persist=True) as user:
+        project_dir = tmp777_path / "test"
+        project = await project_factory(user, project_dir=str(project_dir))
+        payload = {"name": "WF", "project_id": project.id}
+        res = await client.post("api/v1/workflow/", json=payload)
+        workflow = res.json()
+        debug(workflow)
+        assert res.status_code == 201
+
+        file_in = f"{str(testdata_path)}/test_in.json"
+        file_out = f"{str(testdata_path)}/test_out.json"
+
+        if os.path.exists(file_in):
+            with open(file_in, "r") as f:
+                TEST_JSON = json.load(f)
+        else:
+            TEST_JSON = {"test": "json"}
+            with open(file_in, "w") as f:
+                json.dump(TEST_JSON, f)
+
+        task_dict = dict(
+            name="non-python",
+            source="custom task",
+            command=(
+                f"./{str(testdata_path)}/issue189.sh "
+                f"--json {file_in} --metadata-out {file_out}"
+            ),
+            input_type="zarr",
+            output_type="zarr",
+        )
+        task_create = TaskCreate(**task_dict)
+        res = await client.post("api/v1/task/", json=task_create.dict())
+        assert res.status_code == 201
+        task = res.json()
+        res = await client.post(
+            f"api/v1/workflow/{workflow['id']}/add-task/",
+            json=dict(task_id=task["id"]),
+        )
+        assert res.status_code == 201
+
+        input_dataset = await dataset_factory(
+            project, name="input", type="zarr", read_only=False
+        )
+        input_dataset_id = input_dataset.id
+        output_dataset = await dataset_factory(
+            project, name="output", type="zarr", read_only=False
+        )
+        output_dataset_id = output_dataset.id
+
+        await resource_factory(path="/tmp", dataset=output_dataset)
+
+        payload = dict(
+            project_id=project.id,
+            input_dataset_id=input_dataset_id,
+            output_dataset_id=output_dataset_id,
+            workflow_id=workflow["id"],
+            overwrite_input=False,
+        )
+        debug(payload)
+        res = await client.post("/api/v1/project/apply/", json=payload)
+        debug(res.json())
+        assert res.status_code == 202
+
```

Comment:
You should add some waiting time here.
Simplest solution: `time.sleep(5)` (just for debugging)
Better solution: `while` loop which calls the `get_job` endpoint several times, until `status="done"`.

Reply:
Or, at least, something similar to:
```python
        res = await client.post("/api/v1/project/apply/", json=payload)
        job_data = res.json()
        debug(job_data)
        assert res.status_code == 202

        res = await client.get(f"{PREFIX}/job/{job_data['id']}")
        assert res.status_code == 200
        job_status_data = res.json()
        debug(job_status_data)
        assert job_status_data["status"] == "done"
```
so that the failure will be more informative.